### ISSUE TITLE
B-table - how to change current-page from js?

### ISSUE THREAD

**Issue by dmitry-ivanov** [rafaelpimpa/buefy#428]
### Overview of the problem

**Buefy** version: 0.6.1
**Vuejs** version: 2.5.3
**OS/Browser**: Any

### Description

According to https://github.com/rafaelpimpa/buefy/issues/386 - it is possible now to set **initial** current-page for `b-table` component.

But how can I set current-page on some js event?
For example, we have paginated list of rows and simple add form on the same page.
I want to set current-page to 1, after successful submit. But I can't see how can I do it.

Tried `.sync`, didn't help.
>:current-page.sync="page"

Thanks!

**Comment by frantic1048** [comment]
I can confirm this.

Currently `currentPage` is used as `prop` one time, so the binding has no effect.

**Comment by frantic1048** [comment]
I've got an idea to fix this, trying...

**Comment by frantic1048** [comment]
You could check this branch to see if the fix works: https://github.com/frantic1048/buefy/tree/428-react-to-currentPage-prop

**Comment by dmitry-ivanov** [comment]
@frantic1048 Hey, thanks for your help here!

Can you plz provide compiled js file?
I'm using Buefy just from CDN on that example I have.

Thanks!

**Comment by frantic1048** [comment]
I've no idea where to put the compiled files ><

To use that branch

1. clone the repo, and run `npm install && npm run build`
2. remove buefy from your own project
3. add the local buefy build to your own project
4 build your own project to test

**Comment by dmitry-ivanov** [comment]
Hey, @frantic1048 !

I've cloned repo, did build, but now I have a bunch of js files at `dist/static/js`.
Which one can I use instead of Buefy in my project?

Can I compile Buefy to just one js file? I'm using Buefy from CDN at my example:
><script src="https://unpkg.com/buefy"></script>

If so, then I can get that compiled file and use it instead of CDN.

Thanks!

**Comment by frantic1048** [comment]
use `lib/index.js` in Buefy directory after building.

**Comment by dmitry-ivanov** [comment]
I don't have such.

Please, take a look on my build log:
```
npm run build

> buefy@0.6.1 build /Applications/MAMP/htdocs/contributing/buefy
> npm run build:lib && npm run build:docs


> buefy@0.6.1 build:lib /Applications/MAMP/htdocs/contributing/buefy
> npm run build:lib:clean && npm run build:lib:js && npm run build:lib:sass && npm run build:lib:autoprefix && npm run build:lib:cssnano


> buefy@0.6.1 build:lib:clean /Applications/MAMP/htdocs/contributing/buefy
> rimraf lib


> buefy@0.6.1 build:lib:js /Applications/MAMP/htdocs/contributing/buefy
> vue build src/index.js --prod --lib Buefy --dist lib


  We are slimming down vue-cli to optimize the initial installation by removing the `vue build` command.
  Check out Poi (https://github.com/egoist/poi) which offers the same functionality!


> buefy@0.6.1 build:lib:sass /Applications/MAMP/htdocs/contributing/buefy
> node-sass --output-style expanded --source-map true --include-path node_modules src/scss/buefy-build.scss lib/buefy.css

Rendering Complete, saving .css file...
Wrote CSS to /Applications/MAMP/htdocs/contributing/buefy/lib/buefy.css
Wrote Source Map to /Applications/MAMP/htdocs/contributing/buefy/lib/buefy.css.map

> buefy@0.6.1 build:lib:autoprefix /Applications/MAMP/htdocs/contributing/buefy
> postcss --use autoprefixer --output lib/buefy.css lib/buefy.css

✔ Finished lib/buefy.css (476ms)

> buefy@0.6.1 build:lib:cssnano /Applications/MAMP/htdocs/contributing/buefy
> postcss --use cssnano --no-map --output lib/buefy.min.css lib/buefy.css

✔ Finished lib/buefy.css (195ms)

> buefy@0.6.1 build:docs /Applications/MAMP/htdocs/contributing/buefy
> node build/build.js

⠋ building for production...
Starting to optimize CSS...
Processing static/css/app.8d7610f53f0eeb321198f77c9cc6c256.css...
Processed static/css/app.8d7610f53f0eeb321198f77c9cc6c256.css, before: 263647, after: 254993, ratio: 96.72%
⠹ building for production...Total precache size is about 1.66 MB for 39 resources.
Hash: b4b9c0fd6a5b289efb94
Version: webpack 2.7.0
Time: 23847ms
                                                  Asset       Size  Chunks                    Chunk Names
                static/js/8.179ba9c0a185a797bd6e.js.map    91.7 kB       8  [emitted]         
                           static/img/buefy.1d65c18.png    11.2 kB          [emitted]         
                    static/js/0.6b68349e0c7b45e0f676.js    73.9 kB       0  [emitted]         
                    static/js/1.a38f8e43def133f4a09c.js    23.3 kB       1  [emitted]         
                    static/js/2.9b4c7c62a38e79c49b54.js      13 kB       2  [emitted]         
                    static/js/3.9ce02436f82d3d0714b4.js    22.7 kB       3  [emitted]         
                    static/js/4.2bbd6bf4a90f5bbe6af0.js    11.4 kB       4  [emitted]         
                    static/js/5.8b9ebcdc2f726474081c.js    12.5 kB       5  [emitted]         
                    static/js/6.53ee8d7a39e8e6bfca16.js    15.8 kB       6  [emitted]         
                    static/js/7.214d073bf4d92f8e3cca.js    16.1 kB       7  [emitted]         
                    static/js/8.179ba9c0a185a797bd6e.js    20.7 kB       8  [emitted]         
                    static/js/9.21f1a7dce57ea021cfc6.js    11.8 kB       9  [emitted]         
                   static/js/10.d8ca569d07cc619e7655.js    17.1 kB      10  [emitted]         
                   static/js/11.d65ae6a96a2959e62d4f.js      13 kB      11  [emitted]         
                   static/js/12.0fadb72fbd1846633605.js    23.8 kB      12  [emitted]         
                   static/js/13.881c6bf8372af923b2a6.js    11.6 kB      13  [emitted]         
                   static/js/14.b7dde3dd0e9228ad542e.js    10.8 kB      14  [emitted]         
                   static/js/15.6df141a7bec04dbc55f8.js    15.2 kB      15  [emitted]         
                   static/js/16.7b7d2209e3de41d296b5.js    13.4 kB      16  [emitted]         
                   static/js/17.abcb177af78c7fc78cb0.js    13.4 kB      17  [emitted]         
                   static/js/18.60de40c76c1b0d32817c.js    9.21 kB      18  [emitted]         
                   static/js/19.365b8e35a7c66631c0eb.js    8.06 kB      19  [emitted]         
                   static/js/20.bdd25917f804b0900f21.js    9.09 kB      20  [emitted]         
                   static/js/21.fcd549e2a1244a74fb2c.js    25.1 kB      21  [emitted]         
                   static/js/22.ee183425f42dc407c614.js    7.04 kB      22  [emitted]         
                   static/js/23.03d4cfb54768f56dd589.js    7.22 kB      23  [emitted]         
                   static/js/24.dae324889b2e8e388118.js    4.79 kB      24  [emitted]         
                   static/js/25.8503a3601870d4e2d846.js    10.5 kB      25  [emitted]         
                   static/js/26.ef7dc1f4322add6fe4b2.js      19 kB      26  [emitted]         
                   static/js/27.fae550d3e3fe838eec3d.js    4.75 kB      27  [emitted]         
                   static/js/28.63b97041b46abcd96a9b.js    5.85 kB      28  [emitted]         
                   static/js/29.0ab0a98f782141ce823c.js     5.6 kB      29  [emitted]         
                   static/js/30.75a6003f6fbd7e65101f.js     5.4 kB      30  [emitted]         
                   static/js/31.3feac9054ccf9d7428ef.js    6.31 kB      31  [emitted]         
                   static/js/32.5f5f9b031b88b30724a1.js    5.61 kB      32  [emitted]         
                   static/js/33.e5bd3c23353adb81b5de.js    2.34 kB      33  [emitted]         
               static/js/vendor.ccf28f2d727fc9d0cc48.js     789 kB      34  [emitted]  [big]  vendor
                  static/js/app.1dcd1f4008ce19a98738.js     101 kB      35  [emitted]         app
             static/js/manifest.02de45abe07f526397bd.js    2.39 kB      36  [emitted]         manifest
    static/css/app.8d7610f53f0eeb321198f77c9cc6c256.css     255 kB      35  [emitted]  [big]  app
                static/js/0.6b68349e0c7b45e0f676.js.map     304 kB       0  [emitted]         
                static/js/1.a38f8e43def133f4a09c.js.map     129 kB       1  [emitted]         
                static/js/2.9b4c7c62a38e79c49b54.js.map    68.5 kB       2  [emitted]         
                static/js/3.9ce02436f82d3d0714b4.js.map    98.2 kB       3  [emitted]         
                static/js/4.2bbd6bf4a90f5bbe6af0.js.map    61.1 kB       4  [emitted]         
                static/js/5.8b9ebcdc2f726474081c.js.map    64.8 kB       5  [emitted]         
                static/js/6.53ee8d7a39e8e6bfca16.js.map    75.1 kB       6  [emitted]         
                static/js/7.214d073bf4d92f8e3cca.js.map    73.9 kB       7  [emitted]         
                     static/img/buefy-light.7df103a.png      11 kB          [emitted]         
                static/js/9.21f1a7dce57ea021cfc6.js.map    61.8 kB       9  [emitted]         
               static/js/10.d8ca569d07cc619e7655.js.map    67.1 kB      10  [emitted]         
               static/js/11.d65ae6a96a2959e62d4f.js.map    62.3 kB      11  [emitted]         
               static/js/12.0fadb72fbd1846633605.js.map     107 kB      12  [emitted]         
               static/js/13.881c6bf8372af923b2a6.js.map      53 kB      13  [emitted]         
               static/js/14.b7dde3dd0e9228ad542e.js.map      52 kB      14  [emitted]         
               static/js/15.6df141a7bec04dbc55f8.js.map    58.7 kB      15  [emitted]         
               static/js/16.7b7d2209e3de41d296b5.js.map    73.3 kB      16  [emitted]         
               static/js/17.abcb177af78c7fc78cb0.js.map    73.4 kB      17  [emitted]         
               static/js/18.60de40c76c1b0d32817c.js.map    43.9 kB      18  [emitted]         
               static/js/19.365b8e35a7c66631c0eb.js.map    39.3 kB      19  [emitted]         
               static/js/20.bdd25917f804b0900f21.js.map    39.3 kB      20  [emitted]         
               static/js/21.fcd549e2a1244a74fb2c.js.map     151 kB      21  [emitted]         
               static/js/22.ee183425f42dc407c614.js.map    34.3 kB      22  [emitted]         
               static/js/23.03d4cfb54768f56dd589.js.map    34.6 kB      23  [emitted]         
               static/js/24.dae324889b2e8e388118.js.map    26.3 kB      24  [emitted]         
               static/js/25.8503a3601870d4e2d846.js.map    42.9 kB      25  [emitted]         
               static/js/26.ef7dc1f4322add6fe4b2.js.map     116 kB      26  [emitted]         
               static/js/27.fae550d3e3fe838eec3d.js.map    21.5 kB      27  [emitted]         
               static/js/28.63b97041b46abcd96a9b.js.map      25 kB      28  [emitted]         
               static/js/29.0ab0a98f782141ce823c.js.map    27.4 kB      29  [emitted]         
               static/js/30.75a6003f6fbd7e65101f.js.map    20.1 kB      30  [emitted]         
               static/js/31.3feac9054ccf9d7428ef.js.map    28.5 kB      31  [emitted]         
               static/js/32.5f5f9b031b88b30724a1.js.map    23.3 kB      32  [emitted]         
               static/js/33.e5bd3c23353adb81b5de.js.map    9.83 kB      33  [emitted]         
           static/js/vendor.ccf28f2d727fc9d0cc48.js.map    3.47 MB      34  [emitted]         vendor
              static/js/app.1dcd1f4008ce19a98738.js.map     714 kB      35  [emitted]         app
static/css/app.8d7610f53f0eeb321198f77c9cc6c256.css.map     426 kB      35  [emitted]         app
         static/js/manifest.02de45abe07f526397bd.js.map    17.3 kB      36  [emitted]         manifest
                                             index.html    7.86 kB          [emitted]         
                            static/img/buefy-banner.png    19.3 kB          [emitted]         
          static/img/icons/apple-touch-icon-114x114.png    13.2 kB          [emitted]         
          static/img/icons/apple-touch-icon-120x120.png    11.2 kB          [emitted]         
          static/img/icons/apple-touch-icon-144x144.png    17.9 kB          [emitted]         
          static/img/icons/apple-touch-icon-152x152.png    21.8 kB          [emitted]         
            static/img/icons/apple-touch-icon-57x57.png    4.32 kB          [emitted]         
            static/img/icons/apple-touch-icon-60x60.png    4.14 kB          [emitted]         
            static/img/icons/apple-touch-icon-72x72.png    5.85 kB          [emitted]         
            static/img/icons/apple-touch-icon-76x76.png    6.46 kB          [emitted]         
                       static/img/icons/favicon-128.png    6.64 kB          [emitted]         
                     static/img/icons/favicon-16x16.png  796 bytes          [emitted]         
                   static/img/icons/favicon-196x196.png    30.7 kB          [emitted]         
                     static/img/icons/favicon-32x32.png    1.62 kB          [emitted]         
                     static/img/icons/favicon-96x96.png    7.45 kB          [emitted]         
                           static/img/icons/favicon.ico    34.5 kB          [emitted]         
                    static/img/placeholder-1280x960.png    35.9 kB          [emitted]         
                     static/img/placeholder-128x128.png    4.02 kB          [emitted]         
                                   static/manifest.json  416 bytes          [emitted]         

  Build complete.

  Tip: built files are meant to be served over an HTTP server.
  Opening index.html over file:// won't work.
```

**Comment by frantic1048** [comment]
weird ... It looks like you are building the buefy site..

@rafaelpimpa need help ><

**Comment by flatanimals** [comment]
You can also extend the Buefy Table component to add the `currentPage` watcher to achieve this functionality:

```html

<-- PagedTable.vue -->
<script>
import BuefyTable from 'buefy/src/components/table/Table.vue';

export default {
  name: 'PagedTable',
  extends: BuefyTable,
  watch: {
    /**
     * When prop currentPage updated
     * react to it
     */
    currentPage(newVal) {
      this.newCurrentPage = newVal
    }
  },
};
</script>
```

Then you can use it just like you would the Buefy Table component. 

```html
<paged-table
      :data="page.data"
      :loading="loading"
      :paginated="true"
      :total="page.meta.total"
      :current-page="page.meta.current_page"
      :per-page="page.meta.per_page"
      :backend-pagination="true"
      @page-change="onFetchPage"
    >
<!--- your slot definitions -->

</paged-table>
```

**Comment by dmitry-ivanov** [comment]
Thanks, @flatanimals !

But I think it should be a part of Buefy, anyway.

**Comment by frantic1048** [comment]
As @flatanimals says, that's what I did in #429

**Comment by dmitry-ivanov** [comment]
@frantic1048 @rafaelpimpa Thanks!

**Comment by romil-officetroops** [comment]
@flatanimals, My buefy version is 0.5.4 and I tried your solution but it is not working for me. I have tried putting current-page as 2 but it still takes 1.

paged-table(
              :data="data",
              :loading="isLoading",
              :per-page="perPage",
              :paginated="true",
              :backend-pagination="true",
              :total="total",
              @page-change="onPageChange",
              :current-page="currentPage"
            )

And I am doing 
     currentPage = this.$route.query.page 
but it still takes 1

**Comment by dmitry-ivanov** [comment]
@romil-officetroops It was added in `0.6.2`:

https://github.com/buefy/buefy/releases/tag/v0.6.2

**Comment by blakevanlan** [comment]
Surprisingly, I'm still running into this issue (running with `v0.8.6` and `v0.8.9`).

I'm attempting to reset the results to the first page when the user sorts a column. I've updated the backend pagination example to demonstrate the issue:
https://codepen.io/blakevanlan/pen/GRgyozx

I inspected the fields for BTable and BPagination and noticed that `newCurrentPage` is somehow not being updated even though `currentPage` prop is changing.

Any thoughts on what this might be? I've looked through the code to see if I can spot the issue but nothing stands out.
